### ISSUE TITLE
mobilenet v3:  TensorRT model not able to obtain performance of PyTorch Model

### ISSUE THREAD

**Issue by rgkannan676** [wang-xinyu/tensorrtx#719]
Hi @aditya-dl 

I am trying to implement the mobilenet v3 tensorRT model.

I checked the PyTorch model (https://github.com/chufei1995/mobilenetv3.pytorch/blob/master/pretrained/mobilenetv3-large-657e7b3d.pth) accuracy in Mini-ImageNet and obtained accuracy of 75.6 %. But TensorRT model for the same dataset gave a low accuracy of 35.1%. 

I followed the preprocessing steps referring to https://github.com/wang-xinyu/tensorrtx/blob/master/unet/unet.cpp. 

```
cv::Mat preprocess_img(cv::Mat& img) {
    int w, h, x, y;
    //int col=img.cols;
	//int row= img.rows;
	//std::srand((unsigned)time(0));
	
	//int random_col = i = (rand()%()+1;
    float r_w = INPUT_W / (img.cols*1.0);
    float r_h = INPUT_H / (img.rows*1.0);
    if (r_h > r_w) {
        w = INPUT_W;
        h = r_w * img.rows;
        x = 0;
        y = (INPUT_H - h) / 2;
    } else {
        w = r_h* img.cols;
        h = INPUT_H;
        x = (INPUT_W - w) / 2;
        y = 0;
    }
    cv::Mat re(h, w, CV_8UC3);
    cv::resize(img, re, re.size(), 0, 0, cv::INTER_CUBIC);
    cv::Mat out(INPUT_H, INPUT_W, CV_8UC3, cv::Scalar(128, 128, 128));
    re.copyTo(out(cv::Rect(x, y, re.cols, re.rows)));    
    return out;

```

```
int fcount = 0;
    std::string fileName = "";
    for (int f = 0; f < (int)file_names.size(); f++) {
        fcount++;
        if (fcount < BATCH_SIZE && f + 1 != (int)file_names.size()) continue;
        for (int b = 0; b < fcount; b++) {
            fileName = file_names[f - fcount + 1 + b];		
            cv::Mat img = cv::imread(std::string(argv[3]) + "/" + fileName);
            if (img.empty()) continue;
            cv::Mat pr_img = preprocess_img(img); // letterbox BGR to RGB
             int i = 0;
            for (int row = 0; row < INPUT_H; ++row) {
                uchar* uc_pixel = pr_img.data + row * pr_img.step;
                for (int col = 0; col < INPUT_W; ++col) {
                    data[b * 3 * INPUT_H * INPUT_W + i] = ((float)uc_pixel[2]/255.0);
                    data[b * 3 * INPUT_H * INPUT_W + i + INPUT_H * INPUT_W] = ((float)uc_pixel[1]/255.0);
                    data[b * 3 * INPUT_H * INPUT_W + i + 2 * INPUT_H * INPUT_W] = ((float)uc_pixel[0]/255.0);
                    uc_pixel += 3;
                    ++i;
                }
            }		
        }

    auto start = std::chrono::high_resolution_clock::now();
    doInference(*context, data, prob, BATCH_SIZE);
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double, std::milli> time_diff = end - start;
    double fps = (1000 * BATCH_SIZE) / time_diff.count();
	
    }

```
Can you please advise what can be the issue in this. If possible please provide the preprocessing steps.

Thanks.

**Comment by rgkannan676** [comment]
@wang-xinyu 

The 'top 5' accuracy for TensorRT model is 57.5%

**Comment by wang-xinyu** [comment]
You can try input all ones, and compare the output value. And then apply your preprocess and compare again.

**Comment by rgkannan676** [comment]
Hi @wang-xinyu ,

Using mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]  in the pre-processing step improved the results.

**Comment by HareshKarnan** [comment]
Hey, the comment next to preprocess_img() function call says the function is converting RGB to BGR, can you tell me where in the function this conversion is being made ?
